elastic/elastic-agent
# Remove mage notice in favour of make notice

## What this PR actually do?
It removes the `mage notice` target in favour of the `make notice` rule. Please read the PR's body and all comments for a full history/context/reasoning.

## What does this PR do?
Fix a race condition when calling `go-licence-detector`. The original code waited for a goroutine to write to `go-licence-detector`'s stdin before starting the command, so the mage target was hanging.

This commit fixes it, adds logging (when verbose set) to the command run as well as improves the error message.

## Why is it important?
`mage notice` is used to update licences, which is required whenever we update or add dependencies.

## Checklist


## Author's Checklist
- [x] There is no `CHANGELOG-developer.next.asciidoc`, should this fix be mentioned on `CHANGELOG.next.asciidoc`?

## How to test this PR locally
Run `mage notice` from the repo root, it won't hang and if there is an error, it will print it out.

## Related issues

- Blocks #689 
- Relates #1107 

~~## Use cases~~
~~## Screenshots~~
~~## Logs~~
